server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java

```diff
@@ -174,37 +174,6 @@ public FloatVectorValues getFloatVectorValues(String field) throws IOException {
             }
             return queryCancellation.isEnabled() ? new ExitableFloatVectorValues(vectorValues, queryCancellation) : vectorValues;
         }
-
-        @Override
-        public void searchNearestVectors(String field, float[] target, KnnCollector collector, Bits acceptDocs) throws IOException {
-            if (queryCancellation.isEnabled() == false) {
-                in.searchNearestVectors(field, target, collector, acceptDocs);
-                return;
-            }
-            // when acceptDocs is null due to no doc deleted, we will instantiate a new one that would
-            // match all docs to allow timeout checking.
-            final Bits updatedAcceptDocs = acceptDocs == null ? new Bits.MatchAllBits(maxDoc()) : acceptDocs;
-            Bits timeoutCheckingAcceptDocs = new Bits() {
-                private static final int MAX_CALLS_BEFORE_QUERY_TIMEOUT_CHECK = 10;
-                private int calls;
-
-                @Override
-                public boolean get(int index) {
-                    if (calls++ % MAX_CALLS_BEFORE_QUERY_TIMEOUT_CHECK == 0) {
-                        queryCancellation.checkCancelled();
-                    }
-
-                    return updatedAcceptDocs.get(index);
-                }
-
-                @Override
-                public int length() {
-                    return updatedAcceptDocs.length();
-                }
-            };
-
-            in.searchNearestVectors(field, target, collector, acceptDocs);
```

Comment:
We should not delete this. The bug is that `in.searchNearestVectors(field, target, collector, acceptDocs);` 

it should be 
```
in.searchNearestVectors(field, target, collector, timeoutCheckingAcceptDocs);
```

Reply:
🤦 sorry about that, I tried to make up for my mindlessness by drying this up properly now at least :)